src/nvim/state.c

```diff
@@ -127,19 +129,25 @@ char *get_mode(void)
     if (State & VREPLACE_FLAG) {
       buf[0] = 'R';
       buf[1] = 'v';
-    } else if (State & REPLACE_FLAG) {
-      buf[0] = 'R';
     } else {
-      buf[0] = 'i';
+      if (State & REPLACE_FLAG) {
+        buf[0] = 'R';
+      } else {
+        buf[0] = 'i';
+      }
+      if (ins_compl_active()) {
+        buf[1] = 'c';
+      } else if (ctrl_x_mode == 1) {
+        buf[1] = 'x';
+      }
     }
-  } else if (State & CMDLINE) {
+  } else if (State & CMDLINE || exmode_active) {
```

Comment:
I think we usually prefer explicit grouping with bitwise operators (as in the Vim patch).

    (State & CMDLINE) || exmode_active